manifests/params.pp

```diff
@@ -27,26 +27,31 @@
     }
   }
 
-  case $::puppetversion {
-    /^3\.\d\.\d$/: {
+  if str2bool($::is_pe) {
+    #Assuming PE 3
+    $gem_provider = 'pe_gem'
+    $group = 'pe-puppet'
+    $user = 'pe-puppet'
+  } else {
+    if $::puppetversion and versioncmp($::puppetversion, '4.0.0') >= 0 {
+      if $::pe_server_version
```

Comment:
I had modled the params after the puppet-hiera module (https://github.com/voxpupuli/puppet-hiera/blob/master/manifests/params.pp) as it is another module that alters the puppet master and supports several versions. I'll take a closer look to see if that structure can be cleaned up.